flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/CustomDistributionITCase.java

```diff
@@ -0,0 +1,230 @@
+		 */
+
+		ExecutionEnvironment env = ExecutionEnvironment.createLocalEnvironment();
+
+		DataSet<Tuple3<Integer, Long, String>> input1 = env.fromElements(
+						new Tuple3<>(1, 5L, "Hi"),
+						new Tuple3<>(1, 11L, "Hello"),
+						new Tuple3<>(2, 3L, "World"),
+						new Tuple3<>(2, 13L, "Hello World"),
+						new Tuple3<>(3, 8L, "Say"),
+						new Tuple3<>(4, 0L, "Why"),
+						new Tuple3<>(4, 2L, "Java"),
+						new Tuple3<>(4, 11L, "Say Hello"),
+						new Tuple3<>(5, 2L, "Hi Java"));
+
+		final TestDataDist2 dist = new TestDataDist2(2);
+
+		env.setParallelism(dist.getParallelism());
+
+		DataSet<Boolean> result = DataSetUtils.partitionByRange(input1.map(new MapFunction<Tuple3<Integer, Long, String>, Tuple3<Integer, Integer, String>>() {
+			@Override
+			public Tuple3<Integer, Integer, String> map(Tuple3<Integer, Long, String> value) throws Exception {
+				return new Tuple3<>(value.f0, value.f1.intValue(), value.f2);
+			}
+		}), dist, 0, 1).mapPartition(new RichMapPartitionFunction<Tuple3<Integer, Integer, String>, Boolean>() {
+			@Override
+			public void mapPartition(Iterable<Tuple3<Integer, Integer, String>> values, Collector<Boolean> out) throws Exception {
+				int partitionIndex = getRuntimeContext().getIndexOfThisSubtask();
+				for (Tuple3<Integer, Integer, String> s : values) {
+					
+					if ((s.f0 > partitionIndex + 1) || ((s.f0 == partitionIndex + 1) && (s.f1 > dist.rightBoundary[partitionIndex]))) {
+							fail("Record was not correctly partitioned: " + s.toString());
+					}
+				}
+			}
+		});
+
+		result.output(new DiscardingOutputFormat());
+		env.execute();
+	}
+
+	/**
+	 * The class is used to do the tests of range partition with one key.
+	 */
+	public static class TestDataDist1 implements DataDistribution {
+
+		private int dim;
+
+		public TestDataDist1() {}
+
+		/**
+		 * Constructor of the customized distribution for range partition.
+		 * @param dim the number of the fields.
+		 */
+		public TestDataDist1(int dim) {
+			this.dim = dim;
+		}
+
+		public int getParallelism() {
+			return 3;
+		}
+
+		@Override
+		public Object[] getBucketBoundary(int bucketNum, int totalNumBuckets) {
+
+			/*
+			for the first test, the boundary is just like : 
+			(0, 7]
+			(7, 14]
+			(14, 21]
+			 */
+			return new Integer[]{(bucketNum + 1) * 7};
+		}
+
+		@Override
+		public int getNumberOfFields() {
+			return this.dim;
+		}
+
+		@Override
+		public TypeInformation[] getKeyTypes() {
+			return new TypeInformation[]{BasicTypeInfo.INT_TYPE_INFO};
+		}
+
+		@Override
+		public void write(DataOutputView out) throws IOException {
+			out.writeInt(this.dim);
+		}
+
+		@Override
+		public void read(DataInputView in) throws IOException {
+			this.dim = in.readInt();
+		}
+	}
+
+	/**
+	 * The class is used to do the tests of range partition with two keys.
+	 */
+	public static class TestDataDist2 implements DataDistribution {
+
+		public int rightBoundary[] = new int[]{6, 4, 9, 1, 2};
+		private int dim;
```

Comment:
`dim` should always be `2`. Remove the field, the constructor, and update `write()` and `read()`.